src/Routing/Exception/DuplicateNamedRouteException.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @since         3.3.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Routing\Exception;
+
+use Cake\Core\Exception\Exception;
+
+/**
+ * Exception raised when a route names used twice.
+ */
+class DuplicateNamedRouteException extends Exception
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    protected $_messageTemplate = 'A named route was found for "%s" that is already used, route names must be unique across your entire application.';
```

Comment:
@jadb maybe,but big message can help to developer even if found that in error.log file
some times developers will added route runtim(when site is run) and big message can  helpfull to that

@markstory please see new commit